erpnext/hr/doctype/payroll_entry/payroll_entry.py

```diff
@@ -231,36 +231,92 @@ def make_accrual_jv_entry(self):
 
 			accounts = []
 			payable_amount = 0
+			allow_per_employee = frappe.db.get_single_value("HR Settings", "allow_salary_accrual_per_employee")
 
 			# Earnings
 			for acc_cc, amount in earnings.items():
 				payable_amount += flt(amount, precision)
-				accounts.append({
-						"account": acc_cc[0],
-						"debit_in_account_currency": flt(amount, precision),
-						"party_type": '',
-						"cost_center": acc_cc[1] or self.cost_center,
-						"project": self.project
-					})
+
+				if (allow_per_employee & frappe.db.get_value("Account", acc_cc[0], "account_type") == "Payable"):
+						slips = frappe.db.get_all("Salary Slip", filters={'payroll_entry': self.name})
+						for slip in slips:
```

Comment:
I think You can comonify the code for earnings and deduction by passing ```args ``` it is quite repetitive.

Reply:
It is a bit repetitive, but the code in each instance is different enough that factoring out into a method with arguments gets messy. I'm happy to give it another shot if you have a specific implementation in mind, but I think the code is actually cleaner this way.